sdk/clientcore/core/src/main/java/io/clientcore/core/util/serializer/JsonSerializer.java

```diff
@@ -11,7 +11,13 @@
 /**
  * Generic interface covering basic JSON serialization and deserialization methods.
  */
-public interface JsonSerializer extends ObjectSerializer {
+public abstract class JsonSerializer extends ObjectSerializer {
```

Comment:
If you're only changing this type from an interface to an abstract class for the sake of having an implementation of `supportsFormat`, have you considered keeping this as an interface and just having a default implementation?

Reply:
I did, but I didn't want to give the ability to override this method as this serializer type is explicitly for JSON. If there are concerns with this change with regards to migration I'm willing to change this back if that simplifies things.